hil/deferred.py

```diff
@@ -82,26 +82,30 @@ def apply_networking():
     The networking server calls this function in a loop, to ensure that all
     pending network operations get processed within a reasonable amount of
     time.  The return value from this function lets the server know whether it
-    should check for new journal entries, or if it should wait.  If this
-    function does work, the server should immediately check again, because new
-    entries might have been added in the meantime.  But, if this function
-    returns immediately, the server should sleep, because there was no time
-    for new entries to be added.  This keeps the networking server from
+    should check for new journal entries immediately, or if it should wait.  If
+    this function does work, the server should immediately check again, because
+    new entries might have been added in the meantime.  But, if this function
+    returns immediately, the server should sleep, because there was no time for
+    new entries to be added.  This keeps the networking server from
     tight-looping.
     """
-    # Get the journal enries
-    actions = model.NetworkingAction.query \
-        .order_by(model.NetworkingAction.id).all()
 
-    if actions == []:
-        # No actions to perform.  Return False immediately.
+    action = model.NetworkingAction.query \
+        .order_by(model.NetworkingAction.id).first()
+    if action is None:
+        # Optimization - avoid opening a switch session if there's no action
```

Comment:
right, so a session is only created when we call `handle_action`. In that case we can remove the oart above the while loop. How would this look like then?

```python
    session = DaemonSession()

    while True:
        action = model.NetworkingAction.query \
            .order_by(model.NetworkingAction.id).first()
        if action is None:
            db.sesssion.commit()
            return False
        session.handle_action(action)
        db.session.delete(action)
        db.session.commit()

    session.close()
    return True
```
Ideally I would want a do-while here, but python doesn't have it so I am using the `while True` and `break on condition` combo.

@henn what do you think?